packages/web/src/pages/chat-page/components/ChatComposer.tsx

```diff
@@ -6,46 +6,81 @@ import {
   useState
 } from 'react'
 
+import { IconArrow, IconButton } from '@audius/stems'
 import cn from 'classnames'
 
+import { TextAreaV2 } from 'components/data-entry/TextAreaV2'
 import { audiusSdk } from 'services/audius-sdk'
 
 import styles from './ChatComposer.module.css'
 
+const messages = {
+  sendMessage: 'Send Message',
+  sendMessagePlaceholder: 'Start a New Message'
+}
+
+const ENTER_KEY = 'Enter'
+
 export type ChatComposerProps = ComponentPropsWithoutRef<'div'> & {
   chatId?: string
 }
 
 export const ChatComposer = (props: ChatComposerProps) => {
   const { chatId } = props
-  const [message, setMessage] = useState('')
-  const handleInputChange = useCallback(
-    (e: ChangeEvent<HTMLInputElement>) => {
-      setMessage(e.target.value)
+  const [value, setValue] = useState('')
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLTextAreaElement>) => {
+      setValue(e.target.value)
     },
-    [setMessage]
+    [setValue]
   )
+
   const handleSubmit = useCallback(
-    async (e: FormEvent) => {
-      e.preventDefault()
-      if (chatId) {
+    async (e?: FormEvent) => {
+      e?.preventDefault()
+      if (chatId && value) {
+        const message = value
+        setValue('')
         const sdk = await audiusSdk()
         await sdk.chats!.message({ chatId, message })
-        setMessage('')
       }
     },
-    [chatId, message, setMessage]
+    [chatId, value, setValue]
   )
+
+  // Submit when pressing enter while not holding shift
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+      if (e.key === ENTER_KEY && !e.shiftKey) {
+        e.preventDefault()
+        handleSubmit()
+      }
+    },
+    [handleSubmit]
+  )
+
   return (
     <div className={cn(styles.root, props.className)}>
       <form className={styles.form} onSubmit={handleSubmit}>
-        <input
-          disabled={!chatId}
+        <TextAreaV2
           className={styles.input}
-          type='text'
-          value={message}
-          onChange={handleInputChange}
-        />
+          placeholder={messages.sendMessagePlaceholder}
+          onKeyDown={handleKeyDown}
+          onChange={handleChange}
+          value={value}
+          maxVisibleRows={3}
+          maxLength={500}
```

Comment:
let's note to ask product what a sensible max length here is

Reply:
added to doc